analyzers/tests/SonarAnalyzer.TestFramework/Build/ProjectBuilder.cs

```diff
@@ -78,11 +81,21 @@ public ProjectBuilder AddDocument(string path)
         var relativePathFromTestCases = testCasesIndex < 0
             ? throw new ArgumentException($"{nameof(path)} must contain '{TestCases}'", nameof(path))
             : fileInfo.FullName.Substring(testCasesIndex + TestCases.Length);
-        return IsExtensionOfSupportedType(fileInfo)
-            ? AddDocument(project, relativePathFromTestCases, File.ReadAllText(fileInfo.FullName, Encoding.UTF8))
-            : throw new ArgumentException($"The file extension '{fileInfo.Extension}' does not match the project language '{project.Language}' nor Razor.", nameof(path));
+
+        if (!IsExtensionOfSupportedType(fileInfo.Extension))
+        {
+            throw new ArgumentException($"The file extension '{fileInfo.Extension}' does not match the project language '{project.Language}' nor Razor.", nameof(path));
+        }
+
+        return IsRazorDocument(fileInfo.Extension)
+            ? AddAdditionalDocument(project, relativePathFromTestCases, File.ReadAllText(fileInfo.FullName, Encoding.UTF8))
+            : AddDocument(project, relativePathFromTestCases, File.ReadAllText(fileInfo.FullName, Encoding.UTF8));
```

Comment:
This is a bit confusing to me, I much prefer the implementation in Autoscan for 2 reasons:
- the `AddDocument` is calling `AddDocument`, but sometimes for razor content files `AddAdditionalDocument`, which is different (even tho the names are similar).
- with this approach, we are required to filter twice the paths for the razor/cshtml files to load them in the EditorConfigGenerator.

I would suggest adding two dedicated methods `AddAdditionalDocuments` and `AddAdditionalDocument`, extracting the logic in some common functionality and keeping the invocations separate. Then in the Verifier.cs `CreateProject` we will filter the files once and call `AddAdditionalDocuments` for razor files (stored in some private field) and then `AddDocuments` for everything else. WDYT?